py3status/modules/arch_updates.py

```diff
@@ -1,120 +1,102 @@
 
 
 class Py3status:
+    """
+    """
     # available configuration parameters
     cache_timeout = 600
-    format = ''
-    hide_if_zero = False
-    include_aur = False
+    check_aur = None
+    check_pacman = None
+    format = '[\?if=pacman&if=!aur PAC: {pacman}][\?if=!pacman&if=aur AUR: {aur}]' + \
+        '[\?if=pacman&if=aur UPD: {pacman}/{aur}]'
 
     def post_config_hook(self):
-        if self.format == '':
-            if not self.include_aur:
-                self.format = FORMAT_PACMAN_ONLY
+        if self.check_pacman is None:
+            if (self.py3.format_contains(self.format, 'pacman') and
+                    self.py3.check_commands('pacman')):
+                self.check_pacman = True
             else:
-                self.format = FORMAT_PACMAN_AND_AUR
-        self.include_aur = self.py3.format_contains(self.format, 'aur')
-        self.include_pacman = self.py3.format_contains(self.format, 'pacman')
-        if self.py3.format_contains(self.format, 'total'):
-            self.include_aur = True
-            self.include_pacman = True
-
-        # check cower installed
-        if self.include_aur and not self.py3.check_commands(['cower']):
-            self.py3.notify_user('cower is not installed cannot check aur')
-            self.include_aur = False
-
-    def check_updates(self):
-        pacman_updates = self._check_pacman_updates()
-        aur_updates = self._check_aur_updates()
-        if aur_updates == '?':
-            total = pacman_updates
-        else:
-            total = pacman_updates + aur_updates
+                self.check_pacman = False
+
+        if self.check_aur is None:
+            if (self.py3.format_contains(self.format, 'aur') and
+                    self.py3.check_commands('cower')):
+                self.check_aur = True
+            else:
+                self.check_aur = False
+
+        if not self.check_pacman and not self.check_aur:
+            raise Exception(STRING_UNAVAILABLE, self.py3.CACHE_FOREVER)
+
+    def arch_updates(self):
+        pacman = self._check_pacman()
+        aur = self._check_aur()
+        total = pacman + aur
+
+        full_text = self.py3.safe_format(
+            self.format, {'pacman': pacman, 'aur': aur, 'total': total}
+        )
 
-        if self.hide_if_zero and total == 0:
-            full_text = ''
-        else:
-            full_text = self.py3.safe_format(
-                self.format,
-                {
-                    'aur': aur_updates,
-                    'pacman': pacman_updates,
-                    'total': total,
-                }
-            )
         return {
             'cached_until': self.py3.time_in(self.cache_timeout),
             'full_text': full_text,
         }
 
-    def _check_pacman_updates(self):
-        """
-        This method will use the 'checkupdates' command line utility
-        to determine how many updates are waiting to be installed via
-        'pacman -Syu'.
-        """
```

Comment:
I think this is a useful comment it partly explains why we don't just use `pacman`